centreon/src/Core/Dashboard/Infrastructure/API/PartialUpdateDashboard/PartialUpdateDashboardController.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+/*
+ * Copyright 2005 - 2023 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\Dashboard\Infrastructure\API\PartialUpdateDashboard;
+
+use Centreon\Application\Controller\AbstractController;
+use Centreon\Domain\Log\LoggerTrait;
+use Core\Application\Common\UseCase\ErrorResponse;
+use Core\Application\Common\UseCase\InvalidArgumentResponse;
+use Core\Dashboard\Application\UseCase\PartialUpdateDashboard\PartialUpdateDashboard;
+use Core\Dashboard\Application\UseCase\PartialUpdateDashboard\PartialUpdateDashboardRequest;
+use Core\Dashboard\Application\UseCase\PartialUpdateDashboard\Request\PanelRequestDto;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Security\Core\Exception\AccessDeniedException;
+
+final class PartialUpdateDashboardController extends AbstractController
+{
+    use LoggerTrait;
+
+    /**
+     * @param int $dashboardId
+     * @param Request $request
+     * @param PartialUpdateDashboard $useCase
+     * @param PartialUpdateDashboardPresenter $presenter
+     *
+     * @throws AccessDeniedException
+     *
+     * @return Response
+     */
+    public function __invoke(
+        int $dashboardId,
+        Request $request,
+        PartialUpdateDashboard $useCase,
+        PartialUpdateDashboardPresenter $presenter,
+    ): Response {
+        $this->denyAccessUnlessGrantedForApiConfiguration();
+
+        try {
+            $useCase($dashboardId, $this->getPartialUpdateDashboardRequest($request), $presenter);
```

Comment:
Warning: this may lead us to believe that the use case can generate exceptions
```suggestion
            $useCase($dashboardId, $this->getPartialUpdateDashboardRequest($request), $presenter);
```